packages/itwin/quantity-formatting/src/components/quantityformat/internal/StationBaseFactorPreset.tsx

```diff
@@ -0,0 +1,106 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as React from "react";
+import type { FormatDefinition } from "@itwin/core-quantity";
+import { Button, ButtonGroup, Label } from "@itwin/itwinui-react";
+import { useTranslation } from "../../../useTranslation.js";
+
+/** Properties of [[StationBaseFactorPreset]] component.
+ * @internal
+ */
+export interface StationBaseFactorPresetProps {
+  formatProps: FormatDefinition;
+  onChange?: (format: FormatDefinition) => void;
+}
+
+/** Component to provide preset station base factor and offset size combinations.
+ * Provides quick selection for common station configurations.
+ * @internal
+ */
+export function StationBaseFactorPreset(props: StationBaseFactorPresetProps) {
+  const { formatProps, onChange } = props;
+  const { translate } = useTranslation();
+
+  const currentBaseFactor = React.useMemo(() => {
+    return (formatProps as any).stationBaseFactor ?? 1;
+  }, [formatProps]);
```

Comment:
No need to use `useMemo` for primitive values like numbers:
```suggestion
  const currentBaseFactor = (formatProps as any).stationBaseFactor ?? 1;
```